test/ServiceRegistry.js

```diff
@@ -1569,6 +1569,163 @@ describe("ServiceRegistry", function () {
+                    [sentinelOwners, agentInstances[0].address, serviceOwnerAddress], nonce, 0, 0);
+                const signMessageData = await safeContracts.safeSignMessage(agentInstances[0], multisig, txHashData, 0);
+                await safeContracts.executeTx(multisig, txHashData, [signMessageData], 0);
+
+                // Updating a service
+                await serviceRegistry.connect(serviceManager).update(serviceOwnerAddress, configHash, [1], [[4, regBond]],
+                    newMaxThreshold, serviceId);
+
+                // Activate agent instance registration
+                await serviceRegistry.connect(serviceManager).activateRegistration(serviceOwnerAddress, serviceId, {value: regDeposit});
+
+                /// Register agent instance
+                await serviceRegistry.connect(serviceManager).registerAgents(operator, serviceId,
+                    [agentInstances[0].address, agentInstances[1].address, agentInstances[2].address, agentInstances[3].address],
+                    [agentId, agentId, agentId, agentId], {value: 4 * regBond});
+
+                // Change the existent multisig owners and threshold in a multisend transaction using the service owner access
+                let callData = [];
+                let txs = [];
+                nonce = await multisig.nonce();
+                // Add the addresses skipping the first one, but keep the threshold the same
+                for (let i = 0; i < agentInstances.length - 1; i++) {
+                    callData[i] = multisig.interface.encodeFunctionData("addOwnerWithThreshold", [agentInstances[i+1].address, 1]);
+                    txs[i] = safeContracts.buildSafeTransaction({to: multisig.address, data: callData[i], nonce: 0});
+                }
+                // Swap the original multisig owner with the first provided agent instance, and change the threshold separately
+                // Note that the prevOwner is the very first added address as it corresponds to the reverse order of added addresses
+                // The order in the gnosis safe multisig is as follows: sentinelOwners => agentInstances[last].address => ... =>
+                // => newOwnerAddresses[1].address => serviceOwnerAddress
+                // If changing just one owner to another, then instead of newOwnerAddresses[1].address we need sentinelOwners address
+                callData.push(multisig.interface.encodeFunctionData("swapOwner", [agentInstances[1].address, serviceOwnerAddress, agentInstances[0].address]));
+                txs.push(safeContracts.buildSafeTransaction({to: multisig.address, data: callData[callData.length - 1], nonce: 0}));
+                // Change threshold
+                callData.push(multisig.interface.encodeFunctionData("changeThreshold", [newMaxThreshold]));
+                txs.push(safeContracts.buildSafeTransaction({to: multisig.address, data: callData[callData.length - 1], nonce: 0}));
```

Comment:
Instead of removing the service owner here, we just swap it with the first agent instance provided, and change threshold separately. See comments for the explanation.